src/http/server_context.rs

```diff
@@ -119,6 +61,92 @@ impl ServerContext {
       cache: ChronoCache::new(),
       grpc_data_loaders: Arc::new(grpc_data_loaders),
       env_vars: Arc::new(env),
+    };
+
+    for def in bp.definitions.iter_mut() {
+      if let Definition::ObjectTypeDefinition(def) = def {
+        for field in &mut def.fields {
+          if let Some(expr) = &field.resolver {
+            field.resolver = ctx.check_resolver_of_field(field, expr);
+          }
+        }
+      }
+    }
+    ctx.schema = bp.to_schema();
+    ctx.blueprint = bp;
+    ctx
+  }
+
+  fn check_resolver_of_field(&mut self, field: &blueprint::FieldDefinition, expr: &Expression) -> Option<Expression> {
+    let bt = self.blueprint.upstream.batch.clone().unwrap_or_default();
+    let universal_http_client = self.universal_http_client.clone();
+    let http2_only_client = self.http2_only_client.clone();
+    if let Expression::Unsafe(expr_unsafe) = &expr {
```

Comment:
can we split some of the arms to separate functions to make it more readable overall?